test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +111,95 @@ func testManualMeteringInstall(
 	assert.NoError(t, err, "capturing logs and uninstalling metering should produce no error")
 }
 
+func createNFSProvisioner(ctx *deployframework.DeployerCtx)(error) {
+// cteate namespace
+	ctx.TargetPodsCount = nonHDFSTargetPodCount
+
+	_, err := ctx.Client.CoreV1().Namespaces().Get(context.Background(), ctx.Namespace, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+	return nil
+}
+	if apierrors.IsNotFound(err) {
+		n := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ctx.Namespace,
+			Labels: map[string]string{
+				"name": ctx.Namespace + "-" + testingNamespaceLabel,
+			},
+		},
+	}
+	_, err = ctx.Client.CoreV1().Namespaces().Create(context.Background(), n, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	ctx.Logger.Debugf("Created the %s namespace", ctx.Namespace)
+}
+
+//decode and create yamls except for persistentvolume
+	nfsManifestFiles := [3]string{
+ 	"/test/testhelpers/nfs_server.yaml",
+	"/test/testhelpers/nfs_service.yaml",
+ 	"/test/testhelpers/nfs_storageclass.yaml",
+	}
+
+for _, filepath := range nfsManifestFiles {
+// err = deploy.DecodeYAMLManifestToObject(filepath.join(repoDir, nfsManifestsDir, nfsServerManifestFilename), ...)
+
+	err := deploy.DecodeYAMLManifestToObject(manifest, &serverobject)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode the %s YAML service manifest: %v", filepath, err)
```

Comment:
This is the wrong return type looking at the function signature (only returns an error).